packages/grafana-runtime/src/services/dashboard-srv/types.ts

```diff
@@ -0,0 +1,123 @@
+import { PanelModel as BaseCorePanelModel } from '@grafana/data';
+import { GridPos } from '@grafana/schema';
+
+// Middle types to glue Public Plugin APIs types with internal Grafana types
+export interface CorePanelModel extends BaseCorePanelModel {
+  gridPos: GridPos;
+  refresh: () => void;
+}
```

Comment:
Would it make sense to replace this with something like `dashboardSrv.refreshPanel(panel: PanelModel)` and `dashboardSrv.getGridPos(panel: PanelModel)`?

Reply:
I rather not. I prefer we keep the hierarchy logic: dashboardSrv -> dashboard (and actions) -> panel (and actions) so we keep the principle of single responsability.

If we try to put these non dashboardSrv related methods inside dashboardSrv we can end up with a bloated API where dashboardSrv is responsible of doing too many things that are not actually responsability dashboardSrv

Reply:
I guess it depends on the needs from the Twinmaker app. Do we have a summary somewhere on how they are using the API today?

Reply:
here's the list of identified APIs https://github.com/grafana/grafana/issues/73562